cpp/src/arrow/c/bridge.cc

```diff
@@ -1320,14 +1324,16 @@ struct ArrayImporter {
 
   Status Visit(const UnionType& type) {
     auto mode = type.mode();
-    RETURN_NOT_OK(CheckNumBuffers(3));
+    if (mode == UnionMode::SPARSE) {
+      RETURN_NOT_OK(CheckNumBuffers(2));
+    } else {
+      RETURN_NOT_OK(CheckNumBuffers(3));
+    }
```

Comment:
@pitrou

Reply:
That looks fine to me. The more nagging question is: is the IPC format changed?
(given that the integration tests weren't changed in this PR, I would assume "no": is it true?)

Reply:
I did not change the IPC format, which has always stored just two buffers in the sparse case. See https://github.com/apache/arrow/pull/7378/files#diff-78fdd845f659c6d03d17716eb6e27cebR316